internal/app/machined/pkg/runtime/v1alpha1/platform/aws/metadata.go

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// MetadataConfig represents a metadata AWS instance.
+type MetadataConfig struct {
+	Hostname     string `json:"hostname,omitempty"`
```

Comment:
are the JSON tags being used?

Reply:
i've added it to import the data to make unit tests.